### PR TITLE
working in fetch function

### PR DESCRIPTION
**Pull request by euskardi**
- Creation of the document randomGet.java
- Addition of the function callServlets in script.js
- Addition of a textbox in the index.html

image

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/randomGet.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that returns HTML that contains the page view count. */
```
Comment (renguyen) [inline portfolio/src/main/java/com/google/sps/servlets/randomGet.java]: Update the comment?

File: portfolio/src/main/java/com/google/sps/servlets/randomGet.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package com.google.sps.servlets;
```
Comment (renguyen) [inline portfolio/src/main/java/com/google/sps/servlets/randomGet.java]: Can we be more specific with the file name? Perhaps RandomNumberServlet.java?

(also note: file names in Java are typically CamelCase, where even the first word is capitalized)

File: portfolio/src/main/webapp/index.html [hunk 3]
```diff
@@ -38,6 +38,10 @@ <h1>Jesús Urquídez Calvo</h1>
     <p>Click here to get a random message:</p>
     <button onclick="addRandomGreeting()">Hello</button>
     <div id="greeting-container"></div>
+    <button onclick="callServlets();">
+      Random Number from 0 to 9
+    </button>
+    <p id="date-container"></p>
```
Comment (renguyen) [inline portfolio/src/main/webapp/index.html:44]: Can you update the ID here and wherever else it's referenced to be more specific, like `random-number-container`?

File: portfolio/src/main/java/com/google/sps/servlets/randomGet.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that returns HTML that contains the page view count. */
+@WebServlet("/random")
```
Comment (renguyen) [inline portfolio/src/main/java/com/google/sps/servlets/randomGet.java]: Same note about specificity, maybe this address can be "/random-number"

**Review by renguyen — COMMENTED** [review]
Nice screenshot in the description! I like how your code is clean + formatted. Some notes:

It would be nice to have a bit more detail in the description. It's not necessarily about what files you changed, but what's the overall feature change? For this PR, you could include:
* Add servlet to return a random number
* Update client code to fetch and display random number from server

I added some comments about specificity with naming: make sure that readers would be able to tell what a variable, file, etc. does by looking at its name!

File: portfolio/src/main/java/com/google/sps/servlets/randomGet.java (hunk 2 above)
Comment (euskardi) [reply portfolio/src/main/java/com/google/sps/servlets/randomGet.java]: Oh yeah I always confuse CamelCase in methods and files

File: portfolio/src/main/webapp/index.html (hunk 3 above)
Comment (euskardi) [reply portfolio/src/main/webapp/index.html:44]: okay

File: portfolio/src/main/java/com/google/sps/servlets/randomGet.java (hunk 4 above)
Comment (euskardi) [reply portfolio/src/main/java/com/google/sps/servlets/randomGet.java]: check

File: portfolio/src/main/java/com/google/sps/servlets/randomGet.java (hunk 1 above)
Comment (euskardi) [reply portfolio/src/main/java/com/google/sps/servlets/randomGet.java]: check

**Comment by euskardi** [comment]
All comments responded

File: portfolio/src/main/webapp/script.js [hunk 5]
```diff
@@ -26,3 +26,10 @@ function addRandomGreeting() {
   const greetingContainer = document.getElementById('greeting-container');
   greetingContainer.innerText = greeting;
 }
+
+async function callServlets() {
```
Comment (renguyen) [inline portfolio/src/main/webapp/script.js]: One last nit! callServlets() is still kind of general and its only purpose is to just fetch a random number, so what about naming this `fetchRandomNumber()`?

File: portfolio/src/main/webapp/script.js [hunk 6]
```diff
@@ -26,3 +26,10 @@ function addRandomGreeting() {
...
+  const responseFromServer = await fetch(/random-number');
```
Comment (renguyen) [inline portfolio/src/main/webapp/script.js]:
```suggestion
  const responseFromServer = await fetch('/random-number');
```

minor cleanup

File: portfolio/src/main/webapp/script.js [hunk 7]
```diff
@@ -26,3 +26,10 @@ function addRandomGreeting() {
...
+  const textFromResponse = await responseFromServer.text();
+  const dateContainer = document.getElementById('random-number-container');
```
Comment (renguyen) [inline portfolio/src/main/webapp/script.js]: rename dateContainer to randomNumberContainer?

Comment (euskardi) [reply portfolio/src/main/webapp/script.js]: yes!

Comment (euskardi) [reply portfolio/src/main/webapp/script.js]: I had to chenge the name in index.html and script.js to work

File: portfolio/src/main/webapp/script.js (hunk 5 above)
Comment (euskardi) [reply portfolio/src/main/webapp/script.js]: no sooner said than done

File: portfolio/src/main/webapp/script.js (hunk 7 above)
Comment (euskardi) [reply portfolio/src/main/webapp/script.js]: change*

Comment (renguyen) [reply portfolio/src/main/webapp/script.js]: Oh, I mean should the var dateContainer here be changed to randomNumberContainer like:

`const randomNumberContainer = document.getElement...`

(not just the HTML element ID)

**Review by renguyen — APPROVED** [review]
Approved! Thanks for making all the small changes!

**Pull request merged**